plugins/ltac/dune

```diff
@@ -3,6 +3,7 @@
  (public_name coq-core.plugins.ltac)
  (synopsis "Coq's LTAC tactic language")
  (modules :standard \ tauto)
+ (modules_without_implementation tacexpr)
```

Comment:
We keep doing this but it is still a hack...

We should adopt the modern convention to use `_intf` files if that's what we mean instead of relying on the workaround.

Reply:
Do you have a pointer to this convention?

Reply:
The mli-only file is a staple of the OCaml design patterns. I don't understand why this is hack, and if it is, it should be the responsibility of dune to handle this gracefully.

Reply:
> Do you have a pointer to this convention?

Yes, for example https://www.craigfe.io/posts/the-intf-trick , there is a lot of discussion around about .mli -only setups.

> The mli-only file is a staple of the OCaml design patterns. I don't understand why this is hack, and if it is, it should be the responsibility of dune to handle this gracefully.

The problem with .mli files is well-understood and is duplication of type declaration. This becomes annoying; also .mli only files cannot be used by the module alias system (hence why dune generates a .ml file in the back, so it can pack the module). Once you need to use `ppx` etc... you cannot live with .mli only anyways. You can do the `FooOps.ml` thing, but it is ugly and prevents useful patterns such as `private`.

Anyways it was not my intention to open a discussion about this in the PR, I was just kinda wondering to myself if this change was desired, but it is not a big deal.